lib/slax_web/websockets/token.ex

```diff
@@ -0,0 +1,288 @@
+
+  def handle_payload(%{
+        "trigger_id" => trigger_id,
+        "type" => "shortcut",
+        "callback_id" => "access_token"
+      }) do
+    view = build_token_view()
+
+    Slack.open_modal(%{trigger_id: trigger_id, view: view})
+  end
+
+  def handle_payload(%{
+        "trigger_id" => trigger_id,
+        "type" => "block_actions",
+        "view" => %{
+          "root_view_id" => view_id
+        }
+      }) do
+    view = build_repo_view()
+
+    Slack.update_modal(%{trigger_id: trigger_id, view: view, view_id: view_id})
+  end
+
+  def handle_payload(%{
+        "trigger_id" => _trigger_id,
+        "type" => "view_submission",
+        "view" => %{
+          "callback_id" => "token_view",
+          "state" => %{
+            "values" => values
+          }
+        }
+      }) do
+    with %{
+           "repos_select" => %{"selected_options" => repos_options},
+           "token_input" => %{"value" => token},
+           "expiration_datepicker" => %{"selected_date" => expiration_date}
+         } <- parse_state_values(values),
+         {:ok, date} <- Date.from_iso8601(expiration_date) do
+      repos_options
+      |> Enum.map(& &1["value"])
+      |> ProjectRepos.add_token_to_repos(token, date)
+    end
+
+    :ok
+  end
+
+  def handle_payload(%{
+        "trigger_id" => _trigger_id,
+        "type" => "view_submission",
+        "view" => %{
+          "callback_id" => "repo_view",
+          "state" => %{
+            "values" => values
+          }
+        }
+      }) do
+    with %{
+           "project_name_input" => %{"value" => nil},
+           "repo_name_input" => repo_name_input,
+           "project_select" => project_select
+         } <-
+           parse_state_values(values) do
+      # don't create project, create repo and attach it to project
+      {org_name, repo_name} = parse_org_and_repo(repo_name_input["value"])
+
+      ProjectRepos.create(%{
+        org_name: org_name,
+        repo_name: repo_name,
+        project_id: String.to_integer(project_select["selected_option"]["value"])
+      })
+    else
+      %{
+        "project_name_input" => %{"value" => project_name},
+        "repo_name_input" => repo_name_input,
+        "project_select" => _project_select
+      } ->
+        # create project with project name, create repo and attach it to project
+        {org_name, repo_name} = parse_org_and_repo(repo_name_input["value"])
+
+        ProjectRepos.create_repo_with_project(%{
+          project_name: project_name,
+          org_name: org_name,
+          repo_name: repo_name
+        })
+    end
+
+    build_token_view()
+  end
+
+  defp parse_state_values(values) do
+    values
+    |> Map.values()
+    |> Enum.reduce(%{}, fn value, acc -> Map.merge(acc, value) end)
+  end
+
+  defp parse_org_and_repo(org_and_repo) do
+    case String.split(org_and_repo, "/") do
+      [org_name, repo_name] ->
+        {org_name, repo_name}
+
+      [repo_name] ->
+        {"revelrylabs", repo_name}
```

Comment:
The default Github org should probably be a config setting or at least centralized in one hard-coded function call ... is this the first occurrence or are there other places as well?